### PR TITLE
(PUP-10945) Update terminology in Puppet Server loaded http code

### PR DESCRIPTION
**Pull request by justinstoller**
Updates master terminology in the HTTP APIs that Puppet Server loads. First commit changes all of them in a backwards compatible way, second commit changes the internal requires and assertions to the preferred terms.

### CODE REVIEW

**Review by mwaggett — APPROVED** [review]

**Comment by mwaggett** [comment]
Not merging yet so we can give Josh and Night's Watch a chance to look.

**Review by Magisus — COMMENTED** [review]
This doesn't _need_ to target 6.x I don't think, for our upgrade plan to work, but is that where the other terminology changes have been landing? Could this target 6.x?

Do you think we need any superficial tests of the `master` stuff to confirm backwards compatibility?

**Comment by justinstoller** [comment]
> Could this target 6.x?

It could, I think. I had assumed new work on this should go toward 7.x only. I can re-target.

> Do you think we need any superficial tests of the `master` stuff to confirm backwards compatibility?

I'm not sure... I changed almost all of the code and left the tests in the first commit to show that they were backwards compatible. But then I changed the requires in `lib/puppet/network/http.rb`, which means a user that requires that file will no longer have the `Master` namespace available they'll have to explicitly load `lib/puppet/network/http/api/master` for that.

I can change that back and then add a test? Would that be preferable? While targeting 6.x?

**Comment by mwaggett** [comment]
I've sort of lost track of where we're meant to land things by default these days, but I think it makes sense to try to get these changes into the 6.x stream if that's where we already started doing the harmful terminology work.

**Comment by Magisus** [comment]
I think since these changes are not user-facing at all, it's less important to get them into 6.x. I wouldn't block on it if it's a mess.

**Comment by Magisus** [comment]
I think backporting this to 6.x doesn't really buy us anything from a compatibility perspective either, because when we move the server to using the new constants in 8, we'll need to have a package-level version constraint on the agent for >7.4, to avoid allowing installs before this was released in the 7 stream. So that, besides the fact that the changes aren't user-facing, makes me think we should just target main.

File: lib/puppet/network/http/api.rb [hunk 1]
```diff
@@ -18,18 +18,22 @@ def self.not_found_upgrade
...
+    Puppet::Network::HTTP::Route.path(server_prefix).
       any.
-      chain(Puppet::Network::HTTP::API::Master::V3.routes,
+      chain(Puppet::Network::HTTP::API::Server::V3.routes,
             Puppet::Network::HTTP::API.not_found)
   end
+
+  def self.master_routes
+    server_routes
+  end
```
Comment (joshcooper) [inline lib/puppet/network/http/api.rb:38]: also possible to use ruby `:alias` so the methods are the same.

Comment (justinstoller) [reply lib/puppet/network/http/api.rb:38]: That's good to know, I wasn't sure if `alias` worked with class methods.

**Pull request merged**